src/languages/php/tuning.md

```diff
@@ -0,0 +1,110 @@
+
+Determining an optimal `opcache.memory_consumption` is a bit harder, unfortunately, as it requires executing code via a web request to get adequate statistics.  Additionally, pushing new code clears the opcache so the statistics will be non-representative for a while.
+
+Add a a PHP file to your project that is web-accessible containing this line:
+
+```php
+<?php print "<pre>" . print_r(opcache_get_status(false), 1) . "</pre>";
+```
+
+Commit and push it.  Then let the site run for a while (10-20 minutes is probably adequate) and visit that page in a browser.  It should show output similar to the following:
+
+``` 
+Array
+(
+    [opcache_enabled] => 1
+    [cache_full] => 
+    [restart_pending] => 
+    [restart_in_progress] => 
+    [memory_usage] => Array
+        (
+            [used_memory] => 30869976
+            [free_memory] => 36238888
+            [wasted_memory] => 0
+            [current_wasted_percentage] => 0
+        )
+
+    [opcache_statistics] => Array
+        (
+            [num_cached_scripts] => 1518
+            [num_cached_keys] => 2599
+            [max_cached_keys] => 32531
+            [hits] => 5133
+            [start_time] => 1529345972
+            [last_restart_time] => 0
+            [oom_restarts] => 0
+            [hash_restarts] => 0
+            [manual_restarts] => 0
+            [misses] => 1580
+            [blacklist_misses] => 0
+            [blacklist_miss_ratio] => 0
+            [opcache_hit_rate] => 76.463578131983
+        )
+)
+```
+
+The most important values for now are the `used_memory`, `free_memory`, and `oom_restarts` (Out Of Memory Restarts).  If the `oom_restarts` number is high (meaning more than a handful) it means you don't have enough memory allocated to the opcache.  In this example the opcache is using about half of the 64 MB given to it by default, which is fine.  If `free_memory` is too low or `oom_restarts` too high, set a higher value for the memory consumption.
+
+When you're done, be sure to remove the test script we created before.
+
+Your `.platform.app.yaml` file will end up including a block similar to:
+
+```yaml
+variables:
+    php:
+        'opcache.max_accelerated_files': 22000
+        'opcache.memory_consumtion': 96
+```
+
+(Memory consumption is set in megabytes.)
+
+## Optimize your code
+
+It's also possible that your own code is doing more work than it needs to.  Profiling and optimizing a PHP application is a much larger topic than will fit here, but Platform.sh recommends enabling [Blackfire.io](/administration/integrations/blackfire.md) on your project to determine what slow spots can be found and addressed.
```

Comment:
We don't have any guide about using xdebug profiler?

Reply:
No, xdebug's profiler is weak sauce in comparison.  Blackfire is superior in just about every way.  At some point I want to document using xdebug for remote step debugging but trying to document its profiler support is a waste of everyone's time.